databricks/sdk/core.py

```diff
@@ -893,7 +894,29 @@ def __init__(self, cfg: Config = None):
 
         self._session = requests.Session()
         self._session.auth = self._authenticate
-        self._session.mount("https://", HTTPAdapter(max_retries=retry_strategy))
+
+        # Number of urllib3 connection pools to cache before discarding the least
+        # recently used pool. Python requests default value is 10.
+        pool_connections = cfg.connection_pool_size
+        if pool_connections is None:
+            pool_connections = 20
```

Comment:
I understand the need to increase the connection pool size to support concurrent requests. However, why do we need multiple connection pools? Looking at `requests`, HttpAdapter's underlying PoolManager uses the same pool based on the <host,port,scheme> tuple, which for a single client will only be one endpoint.

Reply:
I also do think that we need just one pool, but somehow the default client was using the same value for both.
how about exposing two configuration properties, marking them experimental, and figure out the best defaults later? single SDK client will query only one host.

how should we name `pool_maxsize`? `connection_pool_max`?

Reply:
Default uses the same value for both because it uses the same constant for each parameter as default values in its constructor:
```
    def __init__(
        self,
        pool_connections=DEFAULT_POOLSIZE,
        pool_maxsize=DEFAULT_POOLSIZE,
        max_retries=DEFAULT_RETRIES,
        pool_block=DEFAULT_POOLBLOCK,
    ):
```
`DEFAULT_POOLSIZE` is set to 10.